controllers/etcdcluster_controller.go

```diff
@@ -749,6 +816,23 @@ func (r *EtcdClusterReconciler) reconcile(
 		return result, pdbEvent, nil
 	}
 
+	// GKE requires a resource quota to be made if we're making pods that use
+	// the "system-cluster-critical" or "system-node-critical" priority class.
+	// We use both of these so we create a resource quota per namespace, so that
+	// GKE will schedule our etcd pods
+	quotaExists, err := r.hasResourceQuota(ctx, cluster)
+	if err != nil {
+		return result, nil, fmt.Errorf("unable to fetch resource quota from Kubernetes API: %w", err)
+	}
+	if !quotaExists {
+		quotaCreatedEvent, err := r.createResourceQuota(ctx, cluster)
+		if err != nil {
```

Comment:
Nit: this could ignore apierrors.IsAlreadyExists - can happen if there is a parallel reconciler running.